### ISSUE TITLE
Synced patterns/Reusable blocks: exporting the theme breaks synced patterns

### ISSUE THREAD

**Issue by MaggieCabrera** [WordPress/gutenberg#53806]
### Description

If you build a theme using reusable blocks and then export the theme (both using the core functionality or Create Block theme plugin) the template exported will have the synced patterns as `<!-- wp:block {"ref":62} /-->` which will not work if you try to run the theme on a different environment. We should fix this, even if it's just resolving those refs into the referenced markup of the pattern upon exporting.

### Step-by-step reproduction instructions

Create a synced pattern on one of the theme's templates. 
Export the theme
Try installing that theme on a different environment, you will see what's on the screenshot below

### Screenshots, screen recording, code snippet

<img width="602" alt="Screenshot 2023-08-18 at 11 20 33" src="https://github.com/WordPress/gutenberg/assets/3593343/80a2b4a9-b80e-44ca-bd60-92b1cfc7a3e8">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
I can't find the issue right now, but the Navigation block has the same problem.

This is a general bug with any synced block using database IDs as ref.

**Comment by MaggieCabrera** [comment]
> I can't find the issue right now, but the Navigation block has the same problem.
> 
> This is a general bug with any synced block using database IDs as ref.

true, but for the nav block removing the ID from the block makes it fall back to a list of pages, and the ref is pointing to content. In the case of patterns, we are talking about markup, that is supposed to be part of the theme, not specific to the content of the user's site.

**Comment by Mamaduka** [comment]
Thanks for clarifying, @MaggieCabrera!

**Comment by annezazu** [comment]
Heads up to @WordPress/block-themers!

**Comment by talldan** [comment]
I think the solution for this (same as the navigation block) is for the pattern block to reference patterns using a slug instead of an id. It's discussed in the following two issues:
- https://github.com/WordPress/gutenberg/issues/59272
- https://github.com/WordPress/gutenberg/issues/55911

Though I think there would still be additional work to make sure the referenced synced patterns are exported as files.

**Comment by dballari** [comment]
I just took care of this in a plugin of mine, it could be incorporated into the import / export tool if necessary. One thing that it took me more than 10 years to realize is that the import / export tool tries to keep ids, and referential integrity is only lost when the id is already used by another item. I mentioned it in another issue: https://github.com/WordPress/gutenberg/issues/62566#62686
my solution was to store in an option the ids that where modified (of both navigations and menus) and than perform search replace on the content. Look at lines from 389 to 545 of this [file](https://github.com/dballari/sahmi/blob/main/simply-add-hidden-menu-items.php) . But if the solution is referencing these objects by slug like suggested by @youknowriad then my fix is not needed. I will keep loving WordPress anyway :)
